[docs] Update composition errors docs (#2977)
apollographql/federation: docs/source/errors.mdx

toc: false
---

> When Apollo Federation attempts to compose the schemas provided by your [subgraphs](./building-supergraphs/subgraphs-overview/) into a supergraph schema, it confirms that:

Reviewer: ```suggestion
When GraphOS attempts to compose the schemas provided by your [subgraphs](./building-supergraphs/subgraphs-overview/) into a supergraph schema, it confirms that:
```

Reviewer: As I understand it, Apollo Federation is the architecture, not the thing doing composition. Would GraphOS work here?

Author: Yea, this works better to recognize the entire ecosystem

- The subgraphs have valid GraphQL and Federation syntax
- The resulting supergraph schema is valid

> - The router/gateway has all of the information it needs to execute operations against the resulting schema

Reviewer: ```suggestion
- The graph router or gateway has all of the information it needs to execute operations against the resulting schema
```

> If composition encounters an error, it fails to create the supergraph and will not pick up any subgraph changes. This document lists subgraph validation and composition error codes, along with their root causes.

Reviewer: ```suggestion
If composition fails, a new supergraph schema cannot be generated. This document lists subgraph validation and composition error codes, along with their root causes.
```

<Note>

> Composition might find a schema that technically valid but it can flag potential improvements or hints. These will not throw an error but will be logged as part of the composition build step.
> To learn more, see our docs on [composition hints](/federation/hints/).

Reviewer: ```suggestion

Composition may flag potential improvements or _hints_ on schemas which are technically valid. These hints aren't  errors, and are instead logged as part of the composition build step.
To learn more, see the [composition hints doc](/federation/hints/).
```
